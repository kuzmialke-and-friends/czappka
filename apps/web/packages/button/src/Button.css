button {
    background-color: Transparent;
    background-repeat: no-repeat;
    border: none;
    cursor: pointer;
    overflow: hidden;
    outline: none;
}

p {
    display: none;
}

svg {
    color: var(--color);
}
