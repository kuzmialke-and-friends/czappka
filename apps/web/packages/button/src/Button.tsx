import React from 'react';
import { Icon, IconProps } from './Icon';
import './Button.css';

export interface ButtonProps {
    onClick?: () => void;
    label: string;
    disabled?: boolean;
    icon?: IconProps;
}

export const Button = ({ onClick, label, disabled = false, icon }: ButtonProps) => (
    <button type="button" onClick={onClick} aria-label={label} disabled={disabled}>
        {icon ? <Icon {...icon} /> : null}
        <p> label</p>
    </button>
);
