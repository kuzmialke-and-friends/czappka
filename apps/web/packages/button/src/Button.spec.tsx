import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './Button';

describe('Button', () => {
    it('renders correctly', () => {
        const { container } = render(<Button label="I am a button." />);

        expect(container).toMatchSnapshot();
    });

    it('renders button with icon', () => {
        const { container } = render(<Button label="I have an icon!" />);

        expect(container).toMatchSnapshot();
    });

    it('renders clickable button', () => {
        const onClick = jest.fn();
        const label = 'Click me!';

        render(<Button label={label} onClick={onClick} />);

        fireEvent.click(screen.getByLabelText(label));

        expect(onClick).toHaveBeenCalled();
    });

    it('renders with disabled button', () => {
        const onClick = jest.fn();
        const label = "Can't click me!";

        render(<Button label={label} onClick={onClick} disabled />);

        fireEvent.click(screen.getByLabelText(label));

        expect(onClick).not.toHaveBeenCalled();
    });
});
