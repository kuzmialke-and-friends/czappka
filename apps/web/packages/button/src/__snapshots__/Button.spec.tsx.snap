// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button with icon 1`] = `
<div>
  <button
    aria-label="I have an icon!"
    type="button"
  >
    <p>
       label
    </p>
  </button>
</div>
`;

exports[`Button renders correctly 1`] = `
<div>
  <button
    aria-label="I am a button."
    type="button"
  >
    <p>
       label
    </p>
  </button>
</div>
`;
